lib/Parse/ParseDecl.cpp

```diff
@@ -1481,10 +1481,36 @@ bool Parser::parseDeclAttribute(DeclAttributes &Attributes, SourceLoc AtLoc) {
     return false;
   }
 
+  // FIXME: Remove this after Swift 4.
   if (DK == DAK_Count && Tok.getText() == "NSKeyedArchiverClassName") {
-    DK = DAK_ObjC;
-    diagnose(Tok, diag::attr_nskeyedarchiverclassname_removed)
-      .fixItReplace(Tok.getLoc(), "objc");
+    auto activeDiag = diagnose(Tok,diag::attr_nskeyedarchiverclassname_removed);
+    activeDiag.fixItReplace(Tok.getLoc(), "objc");
```

Comment:
Ah, I see you corrected the Fix-It here that I complained about in the previous PR. Thanks!